eox_core/edxapp_wrapper/backends/users_h_v1.py

```diff
@@ -64,69 +66,70 @@ def create_edxapp_user(*args, **kwargs):
         'activate': True,
         'site': request.site,
         'language_preference': 'es-419',
+        'extended_profile_fields': {
+            'internal_id': '12',
+            ...
+        },
+        'year_of_birth': '1981',
+        'city': 'Boston',
+        'verify_account_email': True,
+        'enable_comments_service': True
     }
     user = create_edxapp_user(**data)
 
     """
     errors = []
-
     email = kwargs.pop("email")
     username = kwargs.pop("username")
     conflicts = check_edxapp_account_conflicts(email=email, username=username)
     if conflicts:
         return None, ["Fatal: account collition with the provided: {}".format(", ".join(conflicts))]
+    request = kwargs.pop("request", False)
+    request.data['name'] = kwargs.pop("fullname")
+    # Get additional settings for this user
+    email_validation = request.data.pop("verify_account_email", False)
+    terms_of_service = request.data.pop("terms_of_service", None)
+    honor_code = request.data.pop("honor_code", None)
+    comments_service = request.data.pop("enable_comments_service", None)
+    extended_profile_data = request.data.pop("extended_profile_fields", {})
+
+    # Settings to be monkey patched to suit request
+    registration_fields = getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
+    site_configuration = configuration_helpers.get_current_site_configuration()
+    # Backup copies of original configurations
+    _original_settings_features = deepcopy(settings.FEATURES)
```

Comment:
I see now, in my comment below I did not understand what you where doing with settings. I would much rather prefer that we use a context_manager to change the settings object instead of modifying it.

Something like what `django.test.utils.override_settings` does but only for the settings.FEATURE dict. override_settings might be too much, but still worth taking a [look](https://github.com/django/django/blob/1.11.20/django/test/utils.py#L395)